### PR TITLE
Use therapist in CI and refactor CI code

### PR DESCRIPTION
**Pull request by rehandalal**
r?

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -18,38 +18,28 @@ dependencies:
...
-    # Save test artifacts for Python and Lint tests
-    - >
-      docker run -v $CIRCLE_TEST_REPORTS:/test_artifacts
-      morgoth:build flake8_junit /test_artifacts/flake8.txt /test_artifacts/flake8.xml
     # Clean up old image and save the new one
-    - mkdir -p ~/cache/docker; test '!' -e ~/cache/docker/image.tar || rm ~/cache/docker/image.tar; docker save morgoth:build > ~/cache/docker/image.tar; ls -l ~/cache/docker
+    - >
+      mkdir -p ~/cache/docker;
+      rm -f ~/cache/docker/image.tar;
+      docker save morgoth:build > ~/cache/docker/image.tar;
```
Comment (mythmon) [inline circle.yml]: Eep, don't do this! I had it wrong in Normandy. The cache is saved at the end of the dependency step, so moving the call to `docker save` here breaks docker caching.

File: circle.yml [hunk 2]
```diff
@@ -18,38 +18,28 @@ dependencies:
...
     - docker info
     # use circleci's docker cache workaround
+    - mkdir -p ~/cache/
     - if [ -e ~/cache/docker/image.tar ]; then echo "Loading image.tar"; docker load -i ~/cache/docker/image.tar || rm ~/cache/docker/image.tar; fi
     # build image
     - docker build -t morgoth:build .
-    # Clean up old image and save the new one
-    - mkdir -p ~/cache/docker; test '!' -e ~/cache/docker/image.tar || rm ~/cache/docker/image.tar; docker save morgoth:build > ~/cache/docker/image.tar; ls -l ~/cache/docker
+    # write the sha256 sum to an artifact to make image verification easier
+    - docker inspect -f '{{.Config.Image}}' morgoth:build | tee $CIRCLE_ARTIFACTS/docker-image-shasum256.txt
```
Comment (mythmon) [inline circle.yml]: Turns out this isn't actually useful to our ppsfolk, so I wouldn't bother adding this (and it should be removed from Normandy).

**Comment by mythmon** [comment]
This has merge conflicts now.

**Comment by rehandalal** [comment]
Rebased to fix merge conflict and updated ^

**Comment by mythmon** [comment]
Huh, sorry I let this sit so long.

**Pull request merged**